pkgs/applications/editors/vim/plugins/patches/aw-watcher-vim/program_paths.patch

```diff
@@ -0,0 +1,13 @@
+diff --git a/plugin/activitywatch.vim b/plugin/activitywatch.vim
+index 6986553..7462b02 100644
+--- a/plugin/activitywatch.vim
++++ b/plugin/activitywatch.vim
+@@ -29,7 +29,7 @@ let s:heartbeat_apiurl = printf('%s/heartbeat?pulsetime=30', s:bucket_apiurl)
+ let s:http_response_code = {}
+ 
+ function! HTTPPostJson(url, data)
+-    let l:req = ['curl', '-s', a:url,
++    let l:req = ['@curl@', '-s', a:url,
```

Comment:
I am still wondering whether we should go as far as patching those...
According to me, it is of the user's responsibility to provide `curl`.

Both solution could work for me but we definitely need to agree on a consistent rule.

Reply:
>  According to me, it is of the user's responsibility to provide `curl`.

Because of Nix sandboxing, some applications that are available in general shell, could not be available in $PATH inside vim. Don't know why, but I saw this issue many times. `curl not found` is a really confusing error to see

> Both solution could work for me but we definitely need to agree on a consistent rule.

Agree, there should be rules. I think [this](https://github.com/NixOS/nixpkgs/pull/334011#discussion_r1764608211) is the right place to discuss.